src/Agent.php

```diff
@@ -881,17 +881,30 @@ public static function cronCleanoldagents($task = null)
                         break;
                     case Conf::STALE_AGENT_ACTION_STATUS:
                         if (isset($config['stale_agents_status']) && $item !== null) {
-                            //change status of agents linked assets
-                            $input = [
-                                'id'        => $item->fields['id'],
-                                'states_id' => $config['stale_agents_status'],
-                                'is_dynamic' => 1
-                            ];
-                            if ($item->update($input)) {
-                                $task->addVolume(1);
-                                $total++;
+                            $shouldUpdate = false;
+                            if (
+                                (isset($config['previous_stale_agents_status']) && $config['previous_stale_agents_status'] != json_encode([0]))
```

Comment:
More easy to read, drop extra useless parenthesis:
```suggestion
                                isset($config['previous_stale_agents_status'])
                                && $config['previous_stale_agents_status'] != json_encode([0])
```